src/content/tools/hooks.md

```diff
@@ -0,0 +1,287 @@
+hook from the  the `hooks` package, which provides a
+Dart API for reading input and writing output.
+
+### Add dependencies
+
+Open your package in a code editor, navigate to its
+`pubspec.yaml`, and then add `hooks` amd `code_assets` as
+`dependencies`. You will also likely need to add a
+tool for your specific task, such as `native_toolchain_c`
+for compiling C code.
+
+:::note
+You need to add the dependencies under `dependencies`, not
+`dev_dependencies`. The hooks will be run by packages and
+applications depending on your package, so the
+Dart code needs to be part of the resolution in those
+packages.
+:::
+
+Example dependencies for a build hook:
+
+```yaml title="pubspec.yaml"
+name: native_add_library
+description: Sums two numbers with native code.
+version: 0.1.0
+
+environment:
+  sdk: '>=3.8.0 <4.0.0'
+
+dependencies:
+  ... 
+  code_assets: any
+  hooks: any
+  native_toolchain_c: any
+
+dev_dependencies:
+  ...
+  ffigen: ^18.0.0
+ ```
+
+### Create a build hook script
+
+If you want to use a build hook, first create the native
+library that you want to use, and then follow these steps to
+connect a build hook to that library in a Dart script: 
+
+1.  Open the script that you want to contain the build hook
+    (for example, `hooks/build.dart`). 
+
+1.  In the main method, add a `build` function from
+    `package:hooks/hooks.dart`. For example:
+
+    ```dart title="hooks/build.dart"
+    import 'package:hooks/hooks.dart';
+    import 'package:logging/logging.dart';
+    import 'package:native_toolchain_c/native_toolchain_c.dart';
+
+    void main(List<String> args) async {
+      await build(args, (input, output) async {
+        final packageName = input.packageName;
+        final cbuilder = CBuilder.library(
+          name: packageName,
+          assetName: '$packageName.dart',
+          sources: ['src/$packageName.c'],
+        );
+        await cbuilder.run(
+          input: input,
+          output: output,
+          logger: Logger('')
+            ..level = Level.ALL
+            ..onRecord.listen((record) => print(record.message)),
+        );
+      });
+    }
+    ```
+
+    * `input`: The input for `hook/build.dart`. This
+      includes the target OS, target architecture,
+      output directory, etc. For details, see
+      the [`BuildInput` class][].
+
+    * `output`: The output for `hook/build.dart`. This
+      includes the assets to be bundled, and dependencies.
+      For details, see the [`BuildOutputBuilder` class][].
+
+### Create a link hook script
+
+If you want to use a link hook, follow these steps to
+add one to a Dart script: 
+
+1.  Open the script that you want to contain the build hook
+    (for example, `hooks/link.dart`). 
+
+1.  In the main method, add a `build` function from
+    `package:hooks/hooks.dart`. For example:
+
+    ```dart title="hooks/link.dart"
+    import 'dart:convert';
+    import 'dart:io';
+
+    import 'package:data_assets/data_assets.dart';
```

Comment:
I don't think we support this yet, correct?

Reply:
Correct.

(Long answer: The data_assets package can be used in the hook during builds. But
the data assets cannot yet be accessed at runtime by the Dart code.)